lib/windshaft/renderers/pg-mvt/renderer.js

```
@@ -25,74 +34,127 @@ module.exports = class PostgresVectorRenderer {
-                if (err.message) {
-                    err.message = "PgMvtRenderer: " + err.message;
-                }
-                return callback(err);
+        const subqueriesPromises = this._getLayerQueries({ z, x, y });
+        const empty_tile_msg = 'Tile does not exist';
+        Promise.all(subqueriesPromises)
+        .then(subqueries =>  {
+            if (subqueries.length === 0) {
+                return callback(new Error(empty_tile_msg));
             }
 
-            const result = data.rows || [];
+            const query = `SELECT ((${subqueries.join(') || (')})) AS st_asmvt`;
+            const timer = new Timer();
+
+            timer.start('query');
+            this.psql.query(query, function (err, data) {
+                timer.end('query');
+                if (err) {
+                    debug("Error running pg-mvt query " + query + ": " + err);
+                    if (err.message) {
+                        err.message = "PgMvtRenderer: " + err.message;
+                    }
+                    return callback(err);
+                }
 
-            if (result.length <= 0 || result[0].st_asmvt === undefined) {
-                return callback(new Error(`Couldn't generate tile`));
-            }
+                const mvt = extractMVT(data);
+                if (!mvt) {
+                    return callback(new Error(empty_tile_msg));
+                }
 
-            const headers = { 'Content-Type': 'application/x-protobuf' };
-            const stats = timer.getTimes();
+                const headers = { 'Content-Type': 'application/x-protobuf' };
+                const stats = timer.getTimes();
 
-            return callback(null, result[0].st_asmvt, headers, stats);
-        });
+                return callback(null, mvt, headers, stats);
+            });
+        }, err => { return callback(err); })
+        .catch(() => { }); // Avoids UnhandledPromiseRejectionWarning
```

If something went wrong in `.all` then you are hiding the underlying error. See the comment above. ^^^